### PR TITLE
Pass lat-lon points to BallTree instead of lon-lat

### PR DESCRIPTION
**Pull request by stelliom**
## Description

The haversine distance metric for *sklearn*'s BallTree requires the points to be passed in lat-lon format (see [docs](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.haversine_distances.html#haversine-distances)). We were passing those in lon-lat format, causing large errors (smoothing/"dripping" artifacts) around 90E and 90W longitude lines.

fr_loam_haversine

Passing the points in lat-lon format seems to solve the issue.

Image

## Workflow for merging PRs

Please read these instructions carefully and follow the steps below before requesting a review by the maintainers. This way we can ensure a smoother review process and your changes will be merged sooner.

Additionally, if this is the first PR you open in EXTPAR make sure to read the [*"Information for EXTPAR Developers"*](https://c2sm.github.io/extpar/development/) section in the documentation.

### Checklist

- [x] Provide a detailed description of your changes in the "Description" section above.
- [x] If you implemented a new feature:
  - [x] Document it in the correct Markdown file(s) under the [`docs/`](https://github.com/C2SM/extpar/tree/master/docs) directory.
  - [x] [Add a new test](https://c2sm.github.io/extpar/testing/#add-a-new-test) or make sure your changes are already tested.
- [x] Your code follows the [style guidelines](https://c2sm.github.io/extpar/development/#coding-rules-and-best-practices).
- [x] Your changes only touch the files/lines relevant for you.
- [x] All four required checks pass (see [*"Testing and debugging"*](#testing-and-debugging) for more details).
- [x] No conflicts with the base branch.

If all the points above are satisfied you can ask for a review by selecting `stelliom` as a reviewer.

For any questions please ping `@stelliom` on this PR.

### Testing and debugging

The most important test for PRs is the one labeled *"EXTPAR Testsuite on Jenkins"*. This checks that the results of all testcases (described by the namelists in [`test/testsuite/data`](https://github.com/C2SM/extpar/tree/master/test/testsuite/data)) did not change compared to the references.

You can launch the testsuite by writing `launch jenkins` as a comment in the PR that you want to test. Once completed, the result of the testsuite will be shown on the PR (failure or success).

If you need more details on the testsuite results (e.g., if you are trying to debug an error or you are simply unsure why the tests fail) you can launch the testsuite with `launch jenkins(debug)`. This will run the tests as usual, but, once completed, you will be given a URL (via a comment on the PR) to access the logfiles and namelists of all tests that were run.

### CODE REVIEW

**Comment by stelliom** [comment]
launch jenkins
